tools/clang/test/DXILValidation/resource_f16.hlsl

```diff
@@ -0,0 +1,15 @@
+// RUN: %dxc -T ps_6_2 -enable-16bit-types %s | FileCheck %s
+
+// CHECK: @main
+
+Texture2D<half>  texture1   : register(t0);
+Texture2D<half4> texture2   : register(t1);
+
+SamplerState samplerState   : register(s0);
+
+half4 main() : SV_TARGET
+{
+    half A = texture1.SampleLevel(samplerState, float2(0.5, 0.5), 0.0);
+    half4 B = texture2.SampleLevel(samplerState, float2(0.5, 0.5), 0.0);
+    return A * B;
+}
```

Comment:
A few comments:
- I think we should move this test to [tools/clang/test/HLSLFileCheck/hlsl/objects/Texture](https://github.com/microsoft/DirectXShaderCompiler/tree/master/tools/clang/test/HLSLFileCheck/hlsl/objects/Texture). 

- It would be good to do a quick scan of the above folder to find existing coverage for `SampleLevel` and possibly combine with this test. E.g., SampleL.hlsl already has some coverage. 
- What about int/uint and min precision types?
- You can use multiple RUN lines in a single file to target different types.

Reply:
SampleLevel is not supported on resources with int/uint types.

I have removed the resource_f16.hlsl test file and added f16 coverage to SampleL.hlsl.

Reply:
Why is this limited to SampleLevel?  It should apply to all Sample intrinsics, shouldn't it?